src/adv.py

```diff
@@ -33,14 +37,80 @@
 room['narrow'].n_to = room['treasure']
 room['treasure'].s_to = room['narrow']
 
+
+
+
 #
 # Main
 #
 
 # Make a new player object that is currently in the 'outside' room.
+new_player= Player("IronMan",room["outside"])
+
 
 # Write a loop that:
-#
+while True:
+    print(new_player)
+
+    print("Enter direction : ")
+    print("North     : n")
+    print("South     : s")
+    print("East      : e")
+    print("West      : w")
+    print("To Quit   : q")
+
+
+    direction = input("Enter the new direction that you want to take :")
+    if direction not in ["n", "s", "e", "w", "q"]:
+        print("Enter valid direction")
+        continue
+
+    #if direction is q, break
+    if direction =="q":
+        print("Game is Over!,Quit the game")
+        break
+
+
+
+    #if direction == "n":
+    # check if room has north direction room.
+    # r is current room : payer.current_room
+    # if r.to_n is not None:
+    # move to r.to_n : player.current_room = r.to_n
+    # else:
+    #  you can't move
+
+    current_room = new_player.current_room
```

Comment:
I like that you've added these test cases to check to see if the next room exists or not